spring-batch-core/src/main/java/org/springframework/batch/core/step/tasklet/SystemCommandTasklet.java

```diff
@@ -142,6 +144,26 @@ else if (stopped) {
 		}
 	}
 
+	/**
+	 * Static setter for the {@link CommandRunner} so it can be adjusted before
+	 * dependency injection. Typically overridden by
+	 * {@link #setCommandRunner(CommandRunner)}.
+	 *
+	 * @param commandRunner {@link CommandRunner} instance to be used by SystemCommandTasklet instance.
+	 */
+	public static void presetCommandRunner(CommandRunner commandRunner) {
+		SystemCommandTasklet.commandRunner = commandRunner;
+	}
+
+	/**
+	 * Injection setter for the {@link CommandRunner}.
+	 *
+	 * @param commandRunner {@link CommandRunner} instance to be used by SystemCommandTasklet instance.
+	 */
+	public void setCommandRunner(CommandRunner commandRunner) {
+		SystemCommandTasklet.commandRunner = commandRunner;
+	}
+
```

Comment:
Here I followed the same pattern used in the `CommandLineJobRunner`, please let me know if there is anything to change about it.